http4s-client-monix-catnap/src/main/scala/com/avast/sst/http4s/client/monix/catnap/Http4sClientCircuitBreakerModule.scala

```diff
@@ -0,0 +1,51 @@
+package com.avast.sst.http4s.client.monix.catnap
+
+import cats.effect.{Clock, Resource, Sync}
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.avast.sst.monix.catnap.CircuitBreakerMetrics.State.{Closed, HalfOpen, Open}
+import com.avast.sst.monix.catnap.{CircuitBreakerConfig, CircuitBreakerMetrics, CircuitBreakerModule}
+import org.http4s.Response
+import org.http4s.client.Client
+
+object Http4sClientCircuitBreakerModule {
+
+  /** Wraps [[org.http4s.client.Client]] with [[monix.catnap.CircuitBreaker]] which provides important metrics and logs breaker events.
+    *
+    * The circuit breaker is special in that it also catches any HTTP 5xx responses and considers them as failures for the circuit breaker.
+    */
+  def make[F[_]: Sync](name: String,
+                       client: Client[F],
+                       circuitBreakerConfig: CircuitBreakerConfig,
+                       circuitBreakerMetrics: CircuitBreakerMetrics[F],
+                       clock: Clock[F]): F[Client[F]] = {
+    val F = Sync[F]
+
+    class ServerFailure(val response: Response[F], val close: F[Unit]) extends Exception
+
+    val cbResource = CircuitBreakerModule[F].makeLoggingCircuitBreaker(
```

Comment:
Why not pass an `CircuitBreaker[F]` as a parameter to `make`, instead of creating it here in place?
(I'm sorry I'm bringing this up again, but I just can't really understand why we can't do that.)

Reply:
It's a bit convoluted because `CircuitBreaker` is not a `trait` and is not straightforward to monitor but I'll try to do something.